operator-framework-core/src/main/java/io/javaoperatorsdk/operator/LeaderElectionManager.java

```diff
@@ -31,24 +38,42 @@ public LeaderElectionManager(ControllerManager controllerManager) {
 
   public void init(LeaderElectionConfiguration config, KubernetesClient client) {
     this.identity = identity(config);
-    Lock lock = new LeaseLock(config.getLeaseNamespace(), config.getLeaseName(), identity);
+    final var leaseNamespace =
+        config.getLeaseNamespace().or(LeaderElectionManager::tryGetInClusterNamespace);
+    if (leaseNamespace.isEmpty()) {
+      final var message =
+          "Lease namespace is not set and cannot be inferred. Leader election cannot continue.";
+      log.error(message);
+      throw new IllegalArgumentException(message);
```

Comment:
Not sure which way is better: throwing an exception to fail or just return so `leaderElector` is still `null`. I prefer to fail loudly, but fine with the other approach.

Reply:
Failing explicitly is better, imo as well.

Reply:
yes, fail loudly that I would prefer too.

Reply:
👍